docs/source/advanced/backends.rst

```diff
@@ -1,14 +1,108 @@
+    | allocBlockSharedMem                        | master thread allocates                       | syncBlockKernels -> master thread allocates -> syncBlockKernels               | syncBlockKernels -> master thread allocates -> syncBlockKernels              | syncBlockKernels -> master thread allocates -> syncBlockKernels                 | syncBlockKernels -> master thread allocates -> syncBlockKernels                                                                    | _shared__                                        |
+    +--------------------------------------------+-----------------------------------------------+-------------------------------------------------------------------------------+------------------------------------------------------------------------------+---------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------+
+    | syncBlockKernels                           | n/a                                           | barrier                                                                       | barrier                                                                      | #pragma omp barrier                                                             | #pragma omp barrier                                                                                                                | _syncthreads                                     |
+    +--------------------------------------------+-----------------------------------------------+-------------------------------------------------------------------------------+------------------------------------------------------------------------------+---------------------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------+--------------------------------------------------+
+    | atomicOp                                   | n/a                                           | std::lockguard< std::mutex >                                                  | n/a                                                                          | #pragma omp critical                                                            | #pragma omp critical                                                                                                               | atomicXXX                                        |
```

Comment:
I would not put it as `n/a`, now a reader may think it's not allowed. While it is allowed, it is just same as normal operations.

Reply:
I wrote `not necessary`